route-rs-runtime/src/link/blackhole_link.rs

```diff
@@ -18,15 +12,29 @@ impl<Packet: Sized> Future for BlackHoleLink<Packet> {
 
     fn poll(&mut self) -> Poll<Self::Item, Self::Error> {
         loop {
-            let input_packet_option: Option<Packet> = try_ready!(self.input_stream.poll());
-            match input_packet_option {
-                None => return Ok(Async::Ready(())),
-                Some(_) => {}
-            };
+            for ingress_stream in self.ingress_streams.iter_mut() {
```

Comment:
Tried using `self.ingress_streams.iter_mut().cycle()` as my iterator, but requires its elements to be `Clone` for some reason. I don't think it's too bad to just keep the loop like this though.

Reply:
There is probably some good reason. I think this loop is okay.